fs-base/src/main/java/com/palantir/giraffe/file/MoreFiles.java

```diff
@@ -248,15 +250,25 @@ public static void copyLarge(Path source, Path target) throws IOException {
      * @param source the path to copy
      * @param target the target path
      *
-     * @throws IllegalArgumentException if the destination already exists or one
-     *         or more parent directories do not exist.
+     * @throws FileAlreadyExistsException if the target already exists
+     * @throws NoSuchFileException if and parents of the target do not exist
      * @throws IOException if an I/O error occurs while copying
      */
     public static void copyRecursive(Path source, Path target) throws IOException {
-        checkArgument(!Files.exists(target),
-                "target (%s) already exists.", target.toAbsolutePath());
-        checkArgument(Files.isDirectory(target.getParent()),
-                "parent directory for target (%s) does not exist.", target.toAbsolutePath());
+        if (Files.exists(target)) {
+            throw new FileAlreadyExistsException(target.toString());
+        }
+
+        if (!Files.exists(target.getParent())) {
+            throw new NoSuchFileException(target.toString());
+        }
+
+        if (target.startsWith(source)) {
```

Comment:
Is this check susceptible to funny business?

what if I do
source = Paths.get(".")
target = source.resolve("foo").toAbsolutePath().normalize()
then
will target.startsWith(source) be true?